crates/bermuda/tests/test_intersection.rs

```diff
@@ -215,3 +209,140 @@ fn test_find_intersections_param(
 ) {
     assert_eq!(intersection::find_intersections(&segments), expected);
 }
+
+#[rstest]
+fn test_find_intersection_points_cross() {
+    /*
+    (1, 0) --- (1, 1)
+        \     /
+         \   /
+          \ /
+           X
+          / \
+         /   \
+        /     \
+    (0, 0) --- (0, 1)
+    */
+    let polygon = vec![vec![
+        Point::new(0.0, 0.0),
+        Point::new(1.0, 1.0),
+        Point::new(1.0, 0.0),
+        Point::new(0.0, 1.0),
+    ]];
+
+    let result = intersection::find_intersection_points(&polygon);
+    let expected = vec![vec![
+        Point::new(0.0, 0.0),
+        Point::new(0.5, 0.5),
+        Point::new(1.0, 1.0),
+        Point::new(1.0, 0.0),
+        Point::new(0.5, 0.5),
+        Point::new(0.0, 1.0),
+    ]];
+
+    assert_eq!(result, expected);
+}
+
+#[rstest]
+fn test_find_intersection_points_cross_intersect_in_point() {
+    /*
+    (1, 0) --- (1, 1)
+        \     /
+         \   /
+          \ /
+           X
+          / \
+         /   \
+        /     \
+    (0, 0) --- (0, 1)
+    */
+    let polygon = vec![vec![
+        Point::new(0.0, 0.0),
+        Point::new(0.5, 0.5),
+        Point::new(1.0, 1.0),
+        Point::new(1.0, 0.0),
+        Point::new(0.0, 1.0),
+    ]];
+
+    let result = intersection::find_intersection_points(&polygon);
+    let expected = vec![vec![
+        Point::new(0.0, 0.0),
+        Point::new(0.5, 0.5),
+        Point::new(1.0, 1.0),
+        Point::new(1.0, 0.0),
+        Point::new(0.5, 0.5),
+        Point::new(0.0, 1.0),
+    ]];
+
+    assert_eq!(result, expected);
+}
+
+#[rstest]
+fn test_find_intersection_points_overleap_edge() {
+    let polygon_list = vec![
+        vec![
+            Point::new(0.0, 0.0),
+            Point::new(3.0, 0.0),
+            Point::new(3.0, 3.0),
+            Point::new(0.0, 3.0),
+        ],
+        vec![
+            Point::new(1.0, 0.0),
+            Point::new(2.0, 0.0),
+            Point::new(2.0, 1.0),
+            Point::new(1.0, 1.0),
+        ],
+    ];
+    let result = intersection::find_intersection_points(&polygon_list);
+    assert_eq!(result.len(), 2);
+    assert_eq!(
+        result[0],
+        vec![
+            Point::new(0.0, 0.0),
+            Point::new(1.0, 0.0),
+            Point::new(2.0, 0.0),
+            Point::new(3.0, 0.0),
+            Point::new(3.0, 3.0),
+            Point::new(0.0, 3.0)
+        ]
+    );
+    assert_eq!(result[1], polygon_list[1]);
+}
+
+#[rstest]
+#[case::rectangle(vec![vec![Point::new(0.0, 0.0), Point::new(1.0, 0.0), Point::new(1.0, 1.0), Point::new(0.0, 1.0)]], 1, 4)]
+#[case::rectangle_in_rectangle(vec![vec![Point::new(0.0, 0.0), Point::new(3.0, 0.0), Point::new(3.0, 3.0), Point::new(0.0, 3.0), Point::new(0.0, 0.0), Point::new(1.0, 1.0),
+    Point::new(2.0, 1.0),
+    Point::new(2.0, 2.0),
+    Point::new(1.0, 2.0),
+    Point::new(1.0, 1.0),]], 2, 8)]
```

Comment:
If possible, it's a bit more readable for the test cases if the spacing is adjusted to use more vertical space.

```
#[case::rectangle_in_rectangle(
    vec![
        vec![
            Point::new(0.0, 0.0),
            Point::new(3.0, 0.0),
            Point::new(3.0, 3.0),
            Point::new(0.0, 3.0),
            Point::new(0.0, 0.0)
        ],
        vec![
            Point::new(1.0, 1.0),
            Point::new(2.0, 1.0),
            Point::new(2.0, 2.0),
            Point::new(1.0, 2.0),
            Point::new(1.0, 1.0)
        ]
    ], 
    2, 
    8
)]
```